aea/agent_loop.py

```diff
@@ -0,0 +1,402 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains the implementation of an agent loop using asyncio."""
+import asyncio
+import datetime
+import logging
+import time
+from abc import ABC, abstractmethod
+from asyncio.events import AbstractEventLoop, TimerHandle
+from asyncio.futures import Future
+from asyncio.tasks import ALL_COMPLETED, FIRST_COMPLETED, Task
+from typing import Any, Callable, Dict, List, Optional, Set, Tuple, Union, cast
+
+from aea.skills.behaviours import TickerBehaviour
```

Comment:
Can we create a `TickerBehaviour` class here which avoids this import?

Then in `aea.skills.behaviours` we should import this class and have `Behaviour(TickerBehaviour, ABC)`

Reply:
done